src/ebay-fake-menu/menu-item.tsx

```diff
@@ -1,40 +1,81 @@
 import React, { ComponentProps, FC, useEffect, useRef } from 'react'
 import classNames from 'classnames'
 import { EbayIcon } from '../ebay-icon'
+import { EbayBadge } from '../ebay-badge'
 
-export type EbayFakeMenuItemProps = Omit<ComponentProps<'a'>, 'onKeyDown' | 'onMouseDown'> & {
-    current?: boolean;
-    disabled?: boolean;
-    autoFocus?: boolean;
-}
+export type EbayMenuItemType = 'button'
+export type EbayFakeMenuItemProps =
+    Omit<ComponentProps<'a'>, 'onKeyDown' | 'onMouseDown'> &
+    Omit<ComponentProps<'button'>, 'onKeyDown' | 'onMouseDown'> &
+    {
+        current?: boolean;
+        disabled?: boolean;
+        autoFocus?: boolean;
+        type?: EbayMenuItemType;
+        badgeNumber?: number;
+        badgeAriaLabel?: string;
+    }
 
 const EbayMenuItem: FC<EbayFakeMenuItemProps> = ({
     className,
     current,
     disabled,
     autoFocus,
+    type,
+    badgeNumber,
+    badgeAriaLabel,
     children,
     ...rest
 }) => {
-    const ref = useRef<HTMLAnchorElement>()
+    const ref = useRef<HTMLAnchorElement & HTMLButtonElement>()
+    const hasBadge = badgeNumber !== undefined
 
     useEffect(() => {
         if (autoFocus) {
             ref.current?.focus()
         }
     })
 
-    return (
+    const itemProps = {
+        ...rest,
+        ref,
+        className: classNames(
+            className,
+            'fake-menu__item',
+            hasBadge && 'menu__item--badged'),
+        'aria-label': badgeAriaLabel
+    }
+
+    const tick = <EbayIcon name="tick16" />
+    const badgeStyleFix = {
+        marginLeft: 'var(--spacing-100)',
+        marginRight: 'var(--spacing-100)'
+    }
+    const badge = hasBadge && <EbayBadge type="menu" number={badgeNumber} style={badgeStyleFix} />
+
+    return type === 'button' ? (
+        <button
+            {...itemProps}
+            type="button"
+            disabled={disabled}
+
+        >
+            <span>
+                {children}
+                {badge}
+            </span>
+            {tick}
+        </button>
+    ) : (
         <a
-            {...rest}
-            ref={ref}
-            className={classNames(className, 'fake-menu__item')}
+            {...itemProps}
             aria-disabled={disabled ? 'true' : undefined}
         >
             <span>
                 {children}
+                {badge}
             </span>
-            <EbayIcon name="tick16" />
+            {tick}
         </a>
     )
```

Comment:
Trying to think of ways we can avoid DRY and make it easier to read . probably a bit of overkill but what do you think 
```
const content = (
    <span>
        {children}
        {badge}
    </span>
);

const Component = type === 'button' ? 'button' : 'a';

const props = type === 'button'
    ? { ...itemProps, type: 'button', disabled }
    : { ...itemProps, 'aria-disabled': disabled ? 'true' : undefined };

return (
    <Component {...props}>
        {content}
        {tick}
    </Component>
);
```

Reply:
I've thought about it, but decided that hurts readability. DRY is good when it's not extreme :)